storybook/figmalinkssource.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <QObject>
+#include <QFileSystemWatcher>
+
+class FigmaLinks;
+
+class FigmaLinksSource : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(QString filePath MEMBER m_filePath WRITE setFilePath NOTIFY filePathChanged)
+    Q_PROPERTY(FigmaLinks* figmaLinks READ getFigmaLinks NOTIFY figmaLinksChanged)
+
+public:
+    explicit FigmaLinksSource(QObject *parent = nullptr);
+
+    void setFilePath(const QString& path);
+    FigmaLinks* getFigmaLinks() const;
+
+signals:
+    void filePathChanged();
+    void figmaLinksChanged();
+
+private:
+    void updateFigmaLinks(const QMap<QString, QStringList>& map);
+    void readFile();
+    void setupWatcher();
+
+    FigmaLinks *m_figmaLinks = nullptr;
+    QString m_filePath;
```

Comment:
```suggestion
    QUrl m_filePath
```
This will save you of a lot of trouble and it's more flexible with the wasm project also. You can remove the cmake SRC_DIR, remove the context property for srcDir and use the json like this:

```
FigmaLinksSource {
         id: figmaMapping

         filePath: "figma.json"
     }
```
     Note: The path needs to be resolved before using FogmaIO and QFileSystemWatcher

Reply:
This comment is gold, thank you!. I don't know why I thought that if I want to use a file from a source path, I have to do it as I did. Of course `QUrl` will do that nicely and will work smoothly with `wasm` config (will look into qrc out of the box).